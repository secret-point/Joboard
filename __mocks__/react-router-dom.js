// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
