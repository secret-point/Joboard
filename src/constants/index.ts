export const IN_PROGRESS = "IN_PROGRESS";
export const PENDING = "PENDING";
export const COMPLETED = "COMPLETED";

export const MAX_MINUTES_FOR_HEARTBEAT = 60;

export const IGNORE_PAGE_TO_STORE_LOCAL: string[] = ["wotc-complete"];

export const CS_DOMAIN_LIST = [
    "https://beta-us.devo.jobsatamazon.hvh.a2z.com",
    "https://gamma-us.devo.jobsatamazon.hvh.a2z.com",
    "https://hiring.amazon.com",
];