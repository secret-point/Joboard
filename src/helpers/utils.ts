import isEmpty from "lodash/isEmpty";

export const convertPramsToJson = (params: string) => {
  if (!isEmpty(params)) {
    return JSON.parse(
      '{"' +
        decodeURI(params.substring(1))
          .replace(/"/g, '\\"')
          .replace(/&/g, '","')
          .replace(/=/g, '":"') +
        '"}'
    );
  } else {
    return {};
  }
};

export const launchAuthentication = () => {
  let hash = window.location.hash.substr(1);

  const origin = window.location.origin;
  const redirectUrl = `${origin}/#${hash}`;

  const state = window.reduxStore.getState();
  let url = `${
    state.app.appConfig.authenticationURL
  }/?redirectUrl=${encodeURIComponent(redirectUrl)}`;
  window.location.assign(url);
};

export const isJson = (obj: string) => {
  try {
    JSON.parse(obj);
  } catch (e) {
    return false;
  }
  return true;
};
