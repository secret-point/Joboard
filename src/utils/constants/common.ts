export const HVH_LOCALE = 'hvh-locale';
