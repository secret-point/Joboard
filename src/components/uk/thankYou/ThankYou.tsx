import React, { useEffect } from "react";
import { ButtonVariant } from "@amzn/stencil-react-components/button";
import { Card } from "@amzn/stencil-react-components/card";
import { FlyoutContent, RenderFlyoutFunctionParams, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { IconArrowRight, IconSize } from "@amzn/stencil-react-components/icons";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import InnerHTML from "dangerously-set-html-content";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  redirectToDashboard,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { CommonColors } from "../../../utils/colors";
import {
  checkAndBoundGetApplication, formatDate, getLocale
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import DebouncedButton from "../../common/DebouncedButton";

interface MapStateToProps {
  application: ApplicationState;
  candidate: CandidateState;
  schedule: ScheduleState;
  job: JobState;
}

export const ThankYou = (props: MapStateToProps) => {
  const { application, candidate, schedule, job } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const applicationData = application.results;
  const { scheduleDetail } = schedule.results;
  const { candidateData } = candidate.results;
  const jobDetail = job.results;
  const nheAppointment = applicationData?.nheAppointment;
  const location = applicationData?.nheAppointment?.location;
  const startTime = nheAppointment?.startTime;
  const endTime = nheAppointment?.endTime;

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    scheduleId && scheduleId!== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleDetail, scheduleId]);

  useEffect(() => {
    if (jobId) {
      jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
    }
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && applicationData && candidateData && scheduleDetail && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData, scheduleDetail, pageName]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, [pageName]);

  const handleGoToDashboard = () => {
    redirectToDashboard();
  };

  const renderNheAppointmentDetails = () => {
    return (
      <>
        <Text fontSize="T300" fontWeight="bold">
          {formatDate(nheAppointment?.dateWithoutFormat, {
            defaultDateFormat: "DD/MM/yyyy",
            displayFormat: "ddd MMM DD"
          })}
        </Text>

        <Text fontSize="T100" fontWeight="bold">
          {location && `${location.streetAddress} ${location.city} ${location.state} ${location.postalCode}`}
        </Text>

        <Text fontSize="T100" color={CommonColors.Neutral70}>
          {t("BB-ThankYou-nhe-appointment-details-visit-us-text", `Visit us for a 30 minute session anytime between ${startTime} - ${endTime}.`, { startTime: startTime, endTime: endTime })}
        </Text>
      </>
    );
  };

  const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-ThankYou-nhe-appointment-flyout-title-text", "Pre-hire appointment")}
      onCloseButtonClick={close}
    >
      <Col width="100%" padding="S300" gridGap={15}>
        <Text fontSize="T200">
          {t("BB-ThankYou-nhe-appointment-flyout-appointment-details-title-text", "Appointment details")}
        </Text>

        {renderNheAppointmentDetails()}

        <Text fontSize="T300">
          {t("BB-ThankYou-nhe-appointment-flyout-things-to-bring-text", "Things to bring")}
        </Text>
        <ul className="ul-list">
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-photoid-text", "Original unexpired photo ID")}
            </Text>
          </li>
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-work-authorization-document-text", "Original unexpired work authorization document")}
            </Text>
          </li>
        </ul>

        <Text fontSize="T100">
          <InnerHTML html={t("BB-Kondo-ThankYou-nhe-appointment-flyout-acceptable-documents-text", "Click <a href='https://glspub.s3-us-west-2.amazonaws.com/Work+Authorization/UK+Employee+Guide/index.html' target='_blank' rel='noopener noreferrer'>here</a> for a complete list of acceptable documents.")} />
        </Text>

        <Text fontSize="T300">
          {t("BB-ThankYou-nhe-appointment-flyout-special-assistance-text", "Special assistance")}
        </Text>
        <Text fontSize="T100">
          <InnerHTML html={t("BB-Kondo-ThankYou-nhe-appointment-flyout-accommodations-assistance-text", "If you have a disability and consider that you may require an adjustment to Amazon's recruitment process, please contact us to discuss on 0808 164 9427 or by email at myapplication@amazon.co.uk")} />
        </Text>
      </Col>
    </FlyoutContent>
  );

  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>
      <Text fontSize="T400">
        {t("BB-ThankYou-thank-you-title-text", "Thank you for applying to Amazon!")}
      </Text>
      <Text fontSize="T200">
        {t("BB-Kondo-ThankYou-description-text", "You will receive an email from us providing further information on the next steps in your application. We look forward to seeing you at your pre-hire appointment.")}
      </Text>

      <Card width="100%" padding="S300" isElevated>
        <Col width="100%" padding="S300" gridGap={15}>
          {renderNheAppointmentDetails()}

          <Row className="border-top-row">
            <WithFlyout renderFlyout={renderFlyout}>
              {({ open }) => (
                <Col alignItems="flex-start">
                  <Text
                    color={CommonColors.Blue70}
                    style={{ cursor: "pointer" }}
                    fontWeight="bold"
                    onClick={open}
                  >
                    <Row
                      gridGap={8}
                      alignItems="center"
                    >
                      {t("BB-Kondo-ThankYou-more-details-text", "More details")}
                      <IconArrowRight size={IconSize.ExtraSmall} aria-hidden />
                    </Row>
                  </Text>
                </Col>
              )}
            </WithFlyout>
          </Row>
        </Col>
      </Card>

      <DebouncedButton variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
        {t("BB-Kondo-ThankYou-return-to-dashboard-text", "Return to dashboard")}
      </DebouncedButton>
    </Col>
  );
};

export const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(ThankYou);
