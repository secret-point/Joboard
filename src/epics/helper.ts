import { log, LoggerType } from "../helpers/log-helper";
import { ProxyApiError } from "../utils/api/types";

export const epicSwitchMapHelper = async (response: any) => {
  if (!response.data || !!response.errorCode || !!response.error || response.errorMessage) {
    log("[Epic] epicSwitchMapHelper: Starting to raise proxy api error", response, LoggerType.ERROR);
    const { errorCode } = response;
    throw createProxyApiEpicError(errorCode, response.error || response.errorMessage);
  }
  return response;
};

export const createProxyApiEpicError = (errorCode: string, errorMessage?: string): ProxyApiError => {
  return { errorCode, errorMessage };
};
