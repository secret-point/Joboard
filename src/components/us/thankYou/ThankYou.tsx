import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Card } from "@amzn/stencil-react-components/card";
import { FlyoutContent, RenderFlyoutFunctionParams, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { DetailedRadio } from "@amzn/stencil-react-components/form";
import { IconArrowRight, IconSize } from "@amzn/stencil-react-components/icons";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import InnerHTML from "dangerously-set-html-content";
import React, { ChangeEvent, useEffect, useState } from "react";
import { connect } from "react-redux";
import { useLocation } from "react-router";
import { addMetricForPageLoad, postAdobeMetrics } from "../../../actions/AdobeActions/adobeActions";
import { boundUpdateApplicationDS } from "../../../actions/ApplicationActions/boundApplicationActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { onCompleteTaskHelper } from "../../../actions/WorkflowActions/workflowActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { ApplicationState } from "../../../reducers/application.reducer";
import { UpdateApplicationRequestDS } from "../../../utils/apiTypes";
import { CommonColors } from "../../../utils/colors";
import { UPDATE_APPLICATION_API_TYPE } from "../../../utils/enums/common";
import {
  checkAndBoundGetApplication,
  createUpdateApplicationRequest,
  formatDate, getLocale,
  validateUserId
} from "../../../utils/helper";
import { Application, FormInputItem } from "../../../utils/types/common";
import FormInputText from "../../common/FormInputText";
import Image from "../../common/Image";
import { translate as t } from "../../../utils/translator";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import {boundResetBannerMessage} from "../../../actions/UiActions/boundUi";
import { METRIC_NAME } from "../../../constants/adobe-analytics";

interface MapStateToProps {
  application: ApplicationState;
  candidate: CandidateState,
  schedule: ScheduleState,
  job: JobState;
}

interface JobReferral {
  hasReferral: 'yes' | 'no';
  referralInfo?: string;
}

export const ThankYou = (props: MapStateToProps) => {
  const { application, candidate, schedule, job } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const applicationData = application.results;
  const scheduleDetail = schedule.results.scheduleDetail;
  const candidateData = candidate.results.candidateData;
  const jobDetail = job.results;
  const nheAppointment = applicationData?.nheAppointment;
  const location = applicationData?.nheAppointment?.location;

  const [hasReferral, setHasReferral] = useState(false);
  const [referralId, setReferralId] = useState("");

  const [formInputConfig, setFormInputConfig] = useState<FormInputItem>({
    hasError: false,
    labelText: "Please provide your referrer login ID (lower case letters only)",
    labelTranslationKey: "BB-ThankYou-referral-login-label-text",
    errorMessage: "Please provide your referrer login ID.",
    errorMessageTranslationKey: "BB-ThankYou-referral-login-empty-error-text",
    required: true,
    name: "referralInfo",
    id: "referral-employee-name",
    dataKey: "jobReferral.referralInfo",
    type: "text"
  });

  useEffect(() => {
    boundGetCandidateInfo();
  }, [])

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    scheduleId && scheduleId!== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    })
  }, [scheduleDetail, scheduleId]);

  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && applicationData && candidateData && scheduleDetail && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData, scheduleDetail, pageName]);

  useEffect(() => {
    return () => {
      //reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    }
  },[pageName])

  const validateFormInput = (): boolean => {
    if (hasReferral) {
      let errorMessage = "";
      let errorMessageTranslationKey = "";

      if (!referralId) {
        errorMessage = "Please provide your referrer login ID.";
        errorMessageTranslationKey = "BB-ThankYou-referral-login-empty-error-text";
      } else if (!validateUserId(referralId)) {
        errorMessage = "User ID should contain only lower case letters and should be at least 4 letters long.";
        errorMessageTranslationKey = "BB-ThankYou-referral-login-regex-error-text";
      }

      if (errorMessage && errorMessageTranslationKey) {
        setFormInputConfig({
          ...formInputConfig,
          hasError: true,
          errorMessage,
          errorMessageTranslationKey
        });

        return false;
      }
    }

    return true;
  };

  const handleGetStarted = () => {
    boundResetBannerMessage();
    if (!validateFormInput()) {
      return;
    }

    if (applicationData) {
      const { THANK_YOU } = UPDATE_APPLICATION_API_TYPE;
      const jobReferral: JobReferral = {
        hasReferral: hasReferral ? "yes" : "no",
      };
      if (hasReferral) {
        jobReferral.referralInfo = referralId;
      }
      const payload = {
        jobReferral,
      };

      const request: UpdateApplicationRequestDS = createUpdateApplicationRequest(applicationData, THANK_YOU, payload);
      boundUpdateApplicationDS(request, (applicationData: Application) => {
        onCompleteTaskHelper(applicationData);
      });

      postAdobeMetrics({name: METRIC_NAME.THANK_YOU_SUBMIT});
    }
  };

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value;
    setReferralId(value);
  };

  const renderNheAppointmentDetails = () => {
    return (
      <>
        <Text fontSize="T300" fontWeight="bold">
          {formatDate(nheAppointment?.dateWithoutFormat, {
            defaultDateFormat: "DD/MM/yyyy",
            displayFormat: "ddd MMM DD"
          })}
        </Text>

        <Text fontSize="T100" fontWeight="bold">
          {location && `${location.streetAddress} ${location.city} ${location.state} ${location.postalCode}`}
        </Text>

        <Text fontSize="T100" color={CommonColors.Neutral70}>
          {t("BB-ThankYou-nhe-appointment-details-visit-us-text", "Visit us for a 30 minute session anytime between {startTime} - {endTime}.", { startTime: nheAppointment?.startTime, endTime: nheAppointment?.endTime })}
        </Text>
      </>
    );
  };

  const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-ThankYou-nhe-appointment-flyout-title-text", "Pre-hire appointment")}
      onCloseButtonClick={close}
    >
      <Col width="100%" padding="S300" gridGap={15}>
        <Text fontSize="T200">
          {t("BB-ThankYou-nhe-appointment-flyout-appointment-details-title-text", "Appointment details")}
        </Text>

        {renderNheAppointmentDetails()}

        <Text fontSize="T300">
          {t("BB-ThankYou-nhe-appointment-flyout-things-to-bring-text", "Things to bring")}
        </Text>
        <ul className="ul-list">
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-photoid-text", "Original unexpired photo ID")}
            </Text>
          </li>
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-work-authorization-document-text", "Original unexpired work authorization document")}
            </Text>
          </li>
        </ul>

        <Text fontSize="T100">
          <InnerHTML html={t("BB-ThankYou-nhe-appointment-flyout-acceptable-documents-text", "Click <a href='https://www.uscis.gov/i-9-central/form-i-9-acceptable-documents' target='_blank' rel='noopener noreferrer'>here</a> for a complete list of acceptable documents.")}/>
        </Text>

        <Text fontSize="T300">
          {t("BB-ThankYou-nhe-appointment-flyout-special-assistance-text", "Special assistance")}
        </Text>
        <Text fontSize="T100">
          <InnerHTML html={t("BB-ThankYou-nhe-appointment-flyout-accommodations-assistance-text", "If you require any accommodations, such as an ASL interpreter for the hiring event, please go to <a href='https://www.amazon.com/accommodations' target='_blank' rel='noopener noreferrer'>www.amazon.com/accommodations</a> and chat with us to request an interpreter. Please contact us immediately upon scheduling your appointment. We need at least 48 hours prior to your appointment to make arrangements.")}/>
        </Text>
      </Col>
    </FlyoutContent >
  );

  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>
      <Image
        src="https://m.media-amazon.com/images/G/01/HVH-Kirin/BB/images/smiley_face.png"
        imgStyles={{ "width": "100%" }}
        aria-hidden="true"
      />
      <Image
        src="https://m.media-amazon.com/images/G/01/HVH-Kirin/BB/images/Icon1.png"
        aria-hidden="true"
        imgStyles={{
          "style": {
            "display": "block",
            "margin": "auto"
          }
        }}
      />

      <Text fontSize="T400" textAlign="center">
        {t("BB-ThankYou-thank-you-title-text", "Thank you for applying to Amazon!")}
      </Text>
      <Text fontSize="T100" textAlign="center">
        {t("BB-ThankYou-look-forward-next-steps-text", "We look forward to seeing you. Here are your next steps:")}
      </Text>

      <Card width="100%" padding="S300" isElevated={true}>
        <Col width="100%" padding="S300" gridGap={15}>
          <Text fontSize="T300" fontWeight="bold">
            {t("BB-ThankYou-referral-details-title-text", "Please provide referral details (if applicable)")}
          </Text>
          <Text fontSize="T200">
            {t("BB-ThankYou-referral-details-description-text", "Were you referred by an existing Amazon employee?")}
          </Text>

          <DetailedRadio
            name="has_referral-radio-col"
            titleText={t("BB-ThankYou-referral-details-yes-text", "Yes")}
            onChange={() => {
              setHasReferral(true);
            }}
          />
          <DetailedRadio
            name="has_referral-radio-col"
            titleText={t("BB-ThankYou-referral-details-no-text", "No")}
            onChange={() => {
              setHasReferral(false);
            }}
          />

          {
            hasReferral &&
            <FormInputText
              inputItem={formInputConfig}
              defaultValue=""
              handleChange={(e: ChangeEvent<HTMLInputElement>) => handleInputChange(e)}
            />
          }

          <Text fontSize="T300" fontWeight="bold">
            {t("BB-ThankYou-start-pre-hire-activities-title-text", "Get started on pre-hire activities")}
          </Text>
          <Text fontSize="T100">
            {t("BB-ThankYou-fill-wotc-description-text", "Before your pre-hire appointment, fill out Work Opportunities Tax Credit Questionnaire.")}
          </Text>

          <Button variant={ButtonVariant.Primary} onClick={handleGetStarted}>
            {t("BB-ThankYou-get-started-button-text", "Let's get started")}
          </Button>
        </Col>
      </Card>

      <Card width="100%" padding="S300" isElevated={true}>
        <Col width="100%" padding="S300" gridGap={15}>
          {renderNheAppointmentDetails()}

          <Row className="border-top-row">
            <WithFlyout renderFlyout={renderFlyout}>
              {({ open }) => (
                <Col alignItems="flex-start">
                  <Text
                    color={CommonColors.Blue70}
                    style={{ cursor: "pointer" }}
                    fontWeight="bold"
                    onClick={open}
                  >
                    <Row
                      gridGap={8}
                      alignItems="center">
                      {t("BB-ThankYou-get-pre-hire-appointment-details-text", "Pre-hire appointment details")}
                      <IconArrowRight size={IconSize.ExtraSmall} />
                    </Row>
                  </Text>
                </Col>
              )}
            </WithFlyout>
          </Row>
        </Col>
      </Card>
    </Col >
  )
}

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(ThankYou);
