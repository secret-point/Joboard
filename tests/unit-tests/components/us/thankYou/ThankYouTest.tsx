import React from "react";
import { shallow } from "enzyme";
import { useLocation } from 'react-router-dom';
import { ThankYou } from "../../../../../src/components/us/thankYou/ThankYou";
import * as helper from '../../../../../src/utils/helper'
import { CountryCode } from "../../../../../src/utils/constants/common";
import { TEST_APPLICATION_ID, TEST_APPLICATION_STATE, TEST_CANDIDATE_STATE, TEST_JOB_ID, TEST_JOB_STATE, TEST_SCHEDULE_ID, TEST_SCHEDULE_STATE, TEST_THANK_YOU_STATE } from "../../../../test-utils/test-data";

describe("ThankYou", () => {
  const mockLocation = {
    pathname: "/thank-you",
    search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}&scheduleId=${TEST_SCHEDULE_ID}`,
    hash: '',
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  // Unit test can't get Katal {{Country}} development value.
  const mockGetCountryCode = jest.spyOn(helper, 'getCountryCode')
  mockGetCountryCode.mockReturnValue(CountryCode.US);

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <ThankYou
        candidate={TEST_CANDIDATE_STATE}
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        schedule={TEST_SCHEDULE_STATE}
        thankYou={TEST_THANK_YOU_STATE}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
