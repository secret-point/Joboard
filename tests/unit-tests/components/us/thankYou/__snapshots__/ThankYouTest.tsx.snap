// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYou should match snapshot 1`] = `
<Col
  gridGap="S300"
  padding={
    Object {
      "top": "S300",
    }
  }
>
  <Image
    aria-hidden="true"
    imgStyles={
      Object {
        "width": "100%",
      }
    }
    src="https://m.media-amazon.com/images/G/01/HVH-Kirin/BB/images/smiley_face.png"
  />
  <Image
    aria-hidden="true"
    imgStyles={
      Object {
        "style": Object {
          "display": "block",
          "margin": "auto",
        },
      }
    }
    src="https://m.media-amazon.com/images/G/01/HVH-Kirin/BB/images/Icon1.png"
  />
  <Text
    font="primary"
    fontSize="T400"
    hyphens="none"
    textAlign="center"
  >
    Thank you for applying to Amazon!
  </Text>
  <Text
    font="primary"
    fontSize="T100"
    hyphens="none"
    textAlign="center"
  >
    We look forward to seeing you. Here are your next steps:
  </Text>
  <Card
    backgroundColor="neutral0"
    display="flex"
    isElevated={true}
    padding="S300"
    status="default"
    width="100%"
  >
    <Col
      gridGap={15}
      padding="S300"
      width="100%"
    >
      <Text
        font="primary"
        fontSize="T300"
        fontWeight="bold"
        hyphens="none"
      >
        Please provide referral details (if applicable)
      </Text>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        Were you referred by an existing Amazon employee?
      </Text>
      <DetailedRadio
        name="has_referral-radio-col"
        onChange={[Function]}
        titleText="Yes"
      />
      <DetailedRadio
        name="has_referral-radio-col"
        onChange={[Function]}
        titleText="No"
      />
      <Text
        font="primary"
        fontSize="T300"
        fontWeight="bold"
        hyphens="none"
      >
        Get started on pre-hire activities
      </Text>
      <Text
        font="primary"
        fontSize="T100"
        hyphens="none"
      >
        Before your pre-hire appointment, fill out Work Opportunities Tax Credit Questionnaire.
      </Text>
      <Button
        colorScheme="default"
        iconPosition="leading"
        onClick={[Function]}
        shouldUseResponsiveContent={true}
        size="default"
        type="button"
        variant="primary"
      >
        Let's get started
      </Button>
    </Col>
  </Card>
  <Card
    backgroundColor="neutral0"
    display="flex"
    isElevated={true}
    padding="S300"
    status="default"
    width="100%"
  >
    <Col
      gridGap={15}
      padding="S300"
      width="100%"
    >
      <Text
        font="primary"
        fontSize="T300"
        fontWeight="bold"
        hyphens="none"
      >
        Thu Aug 04
      </Text>
      <Text
        font="primary"
        fontSize="T100"
        fontWeight="bold"
        hyphens="none"
      >
        Onsite - Recruiting Office at Amazon Distribution Center, 3230 International Place Dupont WA 98327
      </Text>
      <Text
        color="#56616C"
        font="primary"
        fontSize="T100"
        hyphens="none"
      >
        Visit us for a 30 minute session anytime between 13:30 - 14:00.
      </Text>
      <Row
        className="border-top-row"
      >
        <WithFlyout
          flyoutPosition="trailing"
          renderFlyout={[Function]}
        >
          <Component />
        </WithFlyout>
      </Row>
    </Col>
  </Card>
</Col>
`;
